docs/docsite/rst/porting_guides/porting_guide_core_2.19.rst

```diff
@@ -766,6 +766,36 @@ Noteworthy plugin changes
   This filter now returns ``False`` instead of ``None`` when the input is ``None``.
   The aforementioned deprecation warning is also issued in this case.
 
+* Passing nested non-scalars with embedded templates that may resolve to ``Undefined`` to Jinja2
+  filter plugins, such as ``default`` and ``mandatory``, and test plugins including ``defined`` and ``undefined``
+  no longer evaluate as they did in previous versions because nested non-scalars with embedded templates are templated
+  on use only.
+  In 2.19, this assertion passes:
+
+  .. code-block:: yaml
+
+     - assert:
+         that:
+           # Check that the parent variable exists (same as previous versions).
+           - complex_var is defined
```

Comment:
It's not the same as previous versions. It evaluates to False before 2.19, and True after 2.19.

```suggestion
           # Evaluates to False before 2.19, and True after 2.19.
           - complex_var is defined
```